recipes/cppcheck/all/conanfile.py

```diff
@@ -29,11 +29,11 @@ def config_options(self):
         if Version(self.version) >= Version("2.8.0"):
             del self.options.with_z3
 
-    def requirements(self):
+    def build_requirements(self):
         if self.options.get_safe("with_z3", default=False):
-            self.requires("z3/4.8.8")
+            self.tool_requires("z3/4.8.8")
         if self.options.have_rules:
-            self.requires("pcre/8.45")
+            self.tool_requires("pcre/8.45")
```

Comment:
Why should they be build requirements instead of requirements? Looking at source code, pcre is likely is requirements, not a build requirement. I don't know for z3, but it's unlikely a build requirement as well.

Reply:
I'm not sure about that but tool_requires means that these dependencies are only required if the tool is getting built from source. So if I can download a binary package from e.g. conan-center, I don't need these 2 dependencies, because they are statically linked into our executable.
@SpaceIm : Am I right about that?

Reply:
It's not the meaning of build_requires. build_requires is for tools required on build machine in order to build the project, like a compiler, an executable, a script, eventual some configuration files.
z3 & pcre are requirements because cppcheck links to these libraries, so they are required at build time yes, but they must also target the same os/arch, and they may be required at runtime even if they are private (for example if z3 & pcre are shared).